keras/src/layers/preprocessing/image_preprocessing/base_image_preprocessing_layer.py

```diff
@@ -144,17 +158,47 @@ def call(self, data, training=True):
                         "`bounding_box_format='xyxy'`."
                     )
                 bounding_boxes = densify_bounding_boxes(
-                    data["bounding_boxes"], backend=self.backend
+                    data["bounding_boxes"],
+                    is_batched=is_batched,
+                    backend=self.backend,
                 )
+                if "orig_width" not in data:
```

Comment:
What is the implication here -- do users have to specify this key in the data?

Reply:
IMO we should be able to read this information on the input images and perhaps even pass it as part of the `transformation` argument rather than as separate arguments.

Reply:
As we are processing images in batch the images can have varying sizes and the shape of images can have `(None, None, None, 3)`.  So included in broadcasting way by adding extra keys in input dict.

Other way around is basically process single image at a time and then we can have raw image height and width dynamically then we don't need these extra data specific keys. (torchvision Object Detection model follow these). Then batched after this and sent to models.

In this case we have to make sure resize layer with bounding boxes work with only unbatched images and transform boxes at a image level

Reply:
> As we are processing images in batch the images can have varying sizes and the shape of images can have (None, None, None, 3) and bounding_boxes dimensions are respective to raw image

This isn't the case, by the time we hit the layer we are expected dense batches (no ragged tensors). That means all images in the batch have the same size.

Reply:
Oh okay, I was keeping in mind the ragged case as well, so just modified the scenario where if user provides in `ragged`, It will raise the error in transformation. They can use it before the ragged batch and do the resizing without issue. 

I will test now with different scenarios as well so that I don't miss the cases and add them as test cases as well.